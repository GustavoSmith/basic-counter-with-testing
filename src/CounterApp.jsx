import { useState } from "react";

const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  return (
    <main className="flex h-screen flex-col items-center justify-center gap-24 bg-slate-800 text-white">
      <h1 className="text-semibol text-6xl">CounterApp</h1>
      <h2
        aria-label="value"
        className="rounded-full bg-blue-600 px-16 py-2 text-3xl font-semibold text-white"
        /* data-testid="test-value" */
      >
        {count}
      </h2>
      <div className="flex gap-6">
        <button
          aria-label="increase"
          className="rounded-xl border bg-red-500 px-4 text-white"
          onClick={() => setCount(count + 1)}
        >
          +1
        </button>
        <button
          aria-label="decrease"
          className="rounded-xl border bg-red-500 px-4 text-white"
          onClick={() => setCount(count - 1)}
        >
          -1
        </button>
        <button
          aria-label="reset"
          className="rounded-xl border bg-red-500 px-4 text-white"
          onClick={() => setCount(value)}
        >
          Reset
        </button>
      </div>
    </main>
  );
};

export default CounterApp;
