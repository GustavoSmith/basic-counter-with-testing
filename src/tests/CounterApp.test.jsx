import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../CounterApp";

describe("Pruebas de <CounterApp/>", () => {
  const initialValue = 20;

  test("Debe matchear con el snapshot", () => {
    const { container } = render(<CounterApp value={initialValue} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el valor inicial de 100", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();
  });

  test("Debe incrementar con el botón +1", () => {
    render(<CounterApp value={initialValue} />);

    const increaseBtn = screen.getByRole("button", { name: "increase" });

    fireEvent.click(increaseBtn);

    expect(screen.getByText(`${initialValue + 1}`)).toBeTruthy();
  });

  test("Debe decrementar con el botón -1", () => {
    render(<CounterApp value={initialValue} />);

    const decreaseBtn = screen.getByRole("button", { name: "decrease" });

    fireEvent.click(decreaseBtn);

    expect(screen.getByText(`${initialValue - 1}`)).toBeTruthy();
  });

  test("El botón de reset debe resetear el state", () => {
    render(<CounterApp value={initialValue} />);

    const decreaseBtn = screen.getByRole("button", { name: "decrease" });
    const resetBtn = screen.getByRole("button", { name: "reset" });

    fireEvent.click(decreaseBtn);
    fireEvent.click(resetBtn);

    expect(screen.getByText(`${initialValue}`)).toBeTruthy();
  });
});
