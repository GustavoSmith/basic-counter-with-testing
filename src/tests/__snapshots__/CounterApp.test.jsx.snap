// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <CounterApp/> Debe matchear con el snapshot 1`] = `
<div>
  <main
    class="flex h-screen flex-col items-center justify-center gap-24 bg-slate-800 text-white"
  >
    <h1
      class="text-semibol text-6xl"
    >
      CounterApp
    </h1>
    <h2
      aria-label="value"
      class="rounded-full bg-blue-600 px-16 py-2 text-3xl font-semibold text-white"
    >
      20
    </h2>
    <div
      class="flex gap-6"
    >
      <button
        aria-label="increase"
        class="rounded-xl border bg-red-500 px-4 text-white"
      >
        +1
      </button>
      <button
        aria-label="decrease"
        class="rounded-xl border bg-red-500 px-4 text-white"
      >
        -1
      </button>
      <button
        aria-label="reset"
        class="rounded-xl border bg-red-500 px-4 text-white"
      >
        Reset
      </button>
    </div>
  </main>
</div>
`;
